dbt-redshift/tests/functional/adapter/test_exception_handling.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+import threading
+import time
+from dbt.tests.util import run_dbt
+from dbt_common.exceptions import DbtRuntimeError
+
+
+class TestExceptionHandling:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "concurrent_table.sql": """
+                {{ config(materialized='table') }}
+
+                select 1 as id, 'test' as name
+            """
+        }
+
+    def test_concurrent_relation_access_error(self, project, adapter):
```

Comment:
Some comments here suggest this might be a flaky test. Could we add `@pytest.mark.flaky` to this to run it in the flaky suite, which also reruns automatically? I think it will only rerun once, though that can be configured; but it would further reduce the number of false positives we get.